src/particles/diagnostics/DiagnosticOutput.cpp

```diff
@@ -53,34 +55,34 @@ namespace impactx::diagnostics
         int const nLevel = tmp.finestLevel();
         for (int lev = 0; lev <= nLevel; ++lev) {
             // loop over all particle boxes
-            using ParIt = typename decltype(tmp)::ParConstIterType;
-            for (ParIt pti(tmp, lev); pti.isValid(); ++pti) {
+            using MyPinnedParIter = amrex::ParIterSoA<RealSoA::nattribs,IntSoA::nattribs, amrex::PinnedArenaAllocator>;
+            for (MyPinnedParIter pti(tmp, lev); pti.isValid(); ++pti) {
                 const int np = pti.numParticles();
 
                 // preparing access to particle data: AoS
-                using PType = ImpactXParticleContainer::ParticleType;
-                auto const &aos = pti.GetArrayOfStructs();
-                PType const *const AMREX_RESTRICT aos_ptr = aos().dataPtr();
+                //using PType = ImpactXParticleContainer::ParticleType;
+                //auto const &aos = pti.GetArrayOfStructs();
+                //PType const *const AMREX_RESTRICT aos_ptr = aos().dataPtr();
 
                 // preparing access to particle data: SoA of Reals
-                auto &soa_real = pti.GetStructOfArrays().GetRealData();
-                amrex::ParticleReal const *const AMREX_RESTRICT part_px = soa_real[RealSoA::ux].dataPtr();
-                amrex::ParticleReal const *const AMREX_RESTRICT part_py = soa_real[RealSoA::uy].dataPtr();
-                amrex::ParticleReal const *const AMREX_RESTRICT part_pt = soa_real[RealSoA::pt].dataPtr();
+                auto const& particle_attributes = pti.GetStructOfArrays();
+                auto const& part_px = particle_attributes.GetRealData(RealSoA::ux);
+                auto const& part_py = particle_attributes.GetRealData(RealSoA::uy);
+                auto const& part_pt = particle_attributes.GetRealData(RealSoA::pt);
+
+                // Preparing the constructor for the new SoA Particle Type
+                auto ptd = pti.GetParticleTile().getParticleTileData();
```

Comment:
Should this be a reference?
```suggestion
                auto & ptd = pti.GetParticleTile().getParticleTileData();
```